connectors/src/connectors/slack/auto_join_channel.ts

```diff
@@ -0,0 +1,81 @@
+import type { Result } from "@dust-tt/types";
+import { Err, Ok } from "@dust-tt/types";
+
+import { joinChannel } from "@connectors/connectors/slack/lib/channels";
+import { getSlackClient } from "@connectors/connectors/slack/lib/slack_client";
+import {
+  SlackChannel,
+  SlackConfigurationModel,
+} from "@connectors/lib/models/slack";
+import type { Logger } from "@connectors/logger/logger";
+import { ConnectorResource } from "@connectors/resources/connector_resource";
+
+export function isChannelNameWhitelisted(
+  remoteChannelName: string,
+  whiteListedChannelPattern?: string | null
+): boolean {
+  if (!whiteListedChannelPattern || whiteListedChannelPattern === "") {
+    return false;
+  }
+
+  const regex = new RegExp(whiteListedChannelPattern);
+  return regex.test(remoteChannelName);
+}
+
+export async function autoJoinChannel(
+  teamId: string,
+  logger: Logger,
+  requestedConnectorId?: string,
+  slackChannelId?: string
+): Promise<Result<undefined, Error>> {
+  const slackConfiguration = await SlackConfigurationModel.findOne({
+    where: {
+      slackTeamId: teamId,
+    },
+  });
+  if (!slackConfiguration || !slackChannelId) {
+    return new Err(
+      new Error(`Slack configuration not found for teamId ${teamId}`)
+    );
+  }
+  const { connectorId } = slackConfiguration;
+
+  const connector = await ConnectorResource.fetchById(connectorId);
+  if (!connector) {
+    return new Err(new Error(`Connector ${requestedConnectorId} not found`));
```

Comment:
requestedConnectorId is not the one we user to search? What is it for?

Reply:
Indeed, we want to retrieve the connector associated to the `SlackConfiguration`.